src/Ob.cc

```diff
@@ -1135,6 +1135,24 @@ DEF("run",obRun, 1, 2)
 }
 
 
+DEF("run-with-env",obRunWithEnv, 2, 3)
+{
+    Monitor* mon = __CTXT__->monitor;
+    PROTECT(mon);
+
+    CHECK(0, Code, code);
+    if (NARGS == 3) {
```

Comment:
What about calls with other values of NARGS?  Is 2 the only other option?

Reply:
Yeah `DEF("run-with-env",obRunWithEnv, 2, 3)`

means that it can have a minimum of 2 args and maximum of 3